### PR TITLE
Add guildfound functionality

### PR DESCRIPTION
**Pull request by 0xdef1**
Add functionality to allow guild members to set items that they need, and other guild members will see their name in that item's tooltips.

### CODE REVIEW

File: Libs/AceComm-3.0/ChatThrottleLib.lua [hunk 1]
```diff
@@ -0,0 +1,688 @@
...
+-- bandwidth bypassing the library and use less bandwidth itself.
+--
+--
+-- Fully embeddable library. Just copy this file into your addon directory,
+-- add it to the .toc, and it's done.
+--
+-- Can run as a standalone addon also, but, really, just embed it! :-)
+--
+-- LICENSE: ChatThrottleLib is released into the Public Domain
+--
```
Comment (aaronma37) [inline Libs/AceComm-3.0/ChatThrottleLib.lua:24]: Do we need this file? We should have a CTL already

File: GuildFound.lua [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            processMessage(msg, from)
+        end
+    elseif event == "PLAYER_ENTERING_WORLD" then
+        local isLogin, isReload = ...
+        if isLogin or isReload then
+            C_ChatInfo.RegisterAddonMessagePrefix(prefix)
+        end
+    elseif event == "ADDON_LOADED" then
+        local addonName = ...
+        if addonName == "GuildFound" then
```
Comment (aaronma37) [inline GuildFound.lua:82]: Does this work? Wouldn't it still load as OnlyFangs?

File: GuildFound.lua [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+                print("No needers for '" .. item .. "'.")
+            end
+            for k, v in pairs(needers) do
+                print(" - " .. v)
+            end
+        end
+    elseif string.sub(msg, 1, 4) == "help" then
+        prettyPrint("Available commands:")
+        print("|cffffff00/gf help|r print this message")
+        print("|cffffff00/gf add itemName|r add itemName to your list of needed items")
```
Comment (aaronma37) [inline GuildFound.lua:153]: Maybe in a future pr we should add a guild tab with a search bar. Might help for cases where people mispell items. Is it cap sensitive?

File: GuildFound.lua [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    if debug then
+        prettyPrint("|cff40ff40[Debug]|r " .. msg)
+    end
+end
+
+local function sendAddonMessage(msg)
+    if prefix then
+        local msgChannel = "GUILD"
+        lastTx = GetTime()
+        AceComm:SendCommMessage(prefix, msg, msgChannel)
```
Comment (aaronma37) [inline GuildFound.lua:32]: Does acr comms handle messages that are too large?

File: GuildFound.lua [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        print("|cffffff00/gf whoneeds itemName|r print a list of guild members that need itemName")
+    else
+        prettyPrint("Unrecognized command: " .. msg)
+    end
+end
+
+GameTooltip:HookScript("OnTooltipSetItem", function(tooltip, ...)
+    local name = tooltip:GetItem()
+    --Add the name and path of the item's texture
+    local needers = getNeeders(name)
```
Comment (aaronma37) [inline GuildFound.lua:166]: Is this expensive? I'm wondering if we should put the lust generation on a timer instead and maybe disable it in combat

File: OnlyFangs.toc [hunk 6]
```diff
@@ -9,7 +9,7 @@
 
 ## DefaultState: enabled
 ## SavedVariables: OnlyFangsDistributedLog, OnlyFangsStreamerMap, OnlyFangsKeyList, OnlyFangsWeekStart, OnlyFangsRaceInChat, OnlyFangsMonitor, OnlyCSVOut, OnlyMonitorOn, OnlyFangsPrintChatFrame
-## SavedVariablesPerCharacter: OnlyFangsOverrideRace
+## SavedVariablesPerCharacter: OnlyFangsOverrideRace, GuildFound_Saved
```
Comment (aaronma37) [inline OnlyFangs.toc:12]: Would it make sense to have this in the player wide saved variables?

**Review by aaronma37 — APPROVED** [review]

**Comment by aaronma37** [comment]
How big do you expect the saved variables to get? If its a concern, might consider saving them and broadcasting them by itemid

**Comment by aaronma37** [comment]
> How big do you expect the saved variables to get? If its a concern, might consider saving them and broadcasting them by itemid

Though not sure how to get item I'd from just an item name without importing a table, maybe it's not worth

File: Libs/AceComm-3.0/ChatThrottleLib.lua (hunk 1 above)
Comment (0xdef1) [reply Libs/AceComm-3.0/ChatThrottleLib.lua:24]: I'm not sure how AceComm works, was afraid to mess with the files in that folder

File: GuildFound.lua (hunk 2 above)
Comment (0xdef1) [reply GuildFound.lua:82]: oops, need to fix this

File: GuildFound.lua (hunk 3 above)
Comment (0xdef1) [reply GuildFound.lua:153]: case insensitive, yes would love to add some UI but I'm a UI noob in wow :)

File: GuildFound.lua (hunk 4 above)
Comment (0xdef1) [reply GuildFound.lua:32]: yes that's why i'm using it :)

File: GuildFound.lua (hunk 5 above)
Comment (0xdef1) [reply GuildFound.lua:166]: it iterates everyone's wishlist, it might be expensive so yea might be good to do that. how do you tell if you're in combat?

**Comment by 0xdef1** [comment]
> How big do you expect the saved variables to get? If its a concern, might consider saving them and broadcasting them by itemid

Since it's only a single player's wishlist, it won't get very big

**Comment by 0xdef1** [comment]
> > How big do you expect the saved variables to get? If its a concern, might consider saving them and broadcasting them by itemid
> 
> Though not sure how to get item I'd from just an item name without importing a table, maybe it's not worth

Yea I originally thought about using item IDs but it's annoying if you don't already have the item in your bags

File: GuildFound.lua [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+        if not GuildFound_Saved then
+            GuildFound_Saved = {}
+        end
+        GuildFound_Saved.localPlayerNeeds = localPlayerNeeds
+    end
+end
+
+local function OnUpdate()
+    local t = GetTime()
+    if t - lastTx > BROADCAST_INTERVAL then
```
Comment (aaronma37) [inline GuildFound.lua:98]: definitely use a timer here instead

Comment (aaronma37) [reply GuildFound.lua:98]: for context, this event is called every frame

Comment (aaronma37) [reply GuildFound.lua:98]: https://wowwiki-archive.fandom.com/wiki/Using_OnUpdate_correctly

weird its recommended to do something like this on that page though.  IMO use https://wowpedia.fandom.com/wiki/API_C_Timer.NewTimer

File: GuildFound.lua [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+        broadcast()
```
Comment (aaronma37) [inline GuildFound.lua:99]: Should we broadcast if our list is empty?  Just to help lower comms usage

File: OnlyFangs.toc (hunk 6 above)
Comment (aaronma37) [reply OnlyFangs.toc:12]: ohhh I see, so you only see what people need if they are online as well

File: Libs/AceComm-3.0/ChatThrottleLib.lua (hunk 1 above)
Comment (aaronma37) [reply Libs/AceComm-3.0/ChatThrottleLib.lua:24]: I bet its probably fine haha

File: GuildFound.lua (hunk 4 above)
Comment (aaronma37) [reply GuildFound.lua:32]: Nice! Didn't know that. Thats a nice feature

File: GuildFound.lua (hunk 8 above)
Comment (0xdef1) [reply GuildFound.lua]: Yes, it needs to clear the list if items were removed

File: GuildFound.lua (hunk 5 above)
Comment (aaronma37) [reply GuildFound.lua:169]: I think just something like

`UnitAffectingCombat('player')`

File: GuildFound.lua (hunk 7 above)
Comment (0xdef1) [reply GuildFound.lua]: Changed to use a timer

File: WishList.lua [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+        end
+    elseif event == "ADDON_LOADED" then
+        local addonName = ...
+        if addonName == "OnlyFangs" then
+            if not WishList_Saved then
+                WishList_Saved = {}
+            end
+            localPlayerNeeds = WishList_Saved.localPlayerNeeds or {}
+            guildNeeds = WishList_Saved.guildNeeds or {}
+            needersByItem = WishList_Saved.needersByItem or {}
```
Comment (aaronma37) [inline WishList.lua:118]: Does this save everyones data now?

**Pull request merged**